internal/controllers/plans.go

```diff
@@ -209,28 +215,39 @@ func (s Server) AddQuota(ctx echo.Context) error {
 	err = s.GORMDB.Debug().
 		Find(&userPlan, "user_id=?", userID).Error
 	if err != nil {
-		return model.Error(ctx, "user plan name not found for user: "+username, http.StatusInternalServerError)
+		return model.Error(ctx, "user plan name not found for user: "+quotaReq.Username, http.StatusInternalServerError)
 	}
 	userPlanId := *userPlan.ID
 	var quota = model.Quota{
 		UserPlanID:     &userPlanId,
-		Quota:          quotaValueFloat,
+		Quota:          quotaReq.QuotaValue,
 		ResourceTypeID: &resourceID,
 	}
 	err = s.GORMDB.Debug().
+		Clauses(clause.OnConflict{
+			Columns: []clause.Column{
+				{
+					Name: "user_plan_id",
+				},
+				{
+					Name: "resource_type_id",
+				},
+			},
+			DoUpdates: clause.AssignmentColumns([]string{"quota"}),
+		}).
 		Create(&quota).Error
 	if err != nil {
 		return model.Error(ctx, err.Error(), http.StatusInternalServerError)
 	}
 	return model.Success(ctx, "Success", http.StatusOK)
 }
 
-func ParseFloat(valueString string) (float64, error) {
-	valueFloat := 0.0
-	if temp, err := strconv.ParseFloat(valueString, 64); err == nil {
-		valueFloat = temp
-	} else {
-		return valueFloat, err
-	}
-	return valueFloat, nil
-}
+// func ParseFloat(valueString string) (float64, error) {
```

Comment:
If we're no longer using this function, then just go ahead and remove it rather than commenting it out. We can always get it out of git's history if we need it.

If the plan is to uncomment it again later, then a comment explaining that would be useful.